pkg/orchestrator/selection/ranking/previous_executions.go

```diff
@@ -35,15 +35,19 @@ func (s *PreviousExecutionsNodeRanker) RankNodes(ctx context.Context,
 	executions, err := s.jobStore.GetExecutions(ctx, jobstore.GetExecutionsOptions{
 		JobID: job.ID,
 	})
+
 	if err == nil {
 		for _, execution := range executions {
 			if _, ok := previousExecutors[execution.NodeID]; !ok {
 				previousExecutors[execution.NodeID] = 0
 			}
 			previousExecutors[execution.NodeID]++
+
+			// TODO: We do not need this anymore. Double check before we remove it
```

Comment:
I think you are right that we don't need to filter out nodes that have completed the execution successfully before. Also we have a bug where actually we never check `toFilterOut` as we always check for `previousExecutors` first